pkg/frontend/querymiddleware/codec.go

```diff
@@ -402,6 +404,21 @@ func (c prometheusCodec) EncodeRequest(ctx context.Context, r Request) (*http.Re
 	return req.WithContext(ctx), nil
 }
 
+func setRequestHeadersFromOptions(req *http.Request, o Options) {
```

Comment:
the opposite function is called `decodeOptions` ([code](https://github.com/grafana/mimir/blob/9fd2b19a61acec59336d677a50270806dcc4482c/pkg/frontend/querymiddleware/codec.go#L323)). WDYT we name this `encodeOptions`?

Reply:
I'm generally not too fond of the encode/decode naming in this case because (at least for me) it's not obvious which direction of the conversion is encode and which is decode. Happy to change it to your proposal if you think staying consistent trumps that in this case?

Reply:
I'm also happy to change the other method to something like `setOptionsFromRequestHeaders` so we keep it consistent. I was more concerned with being able to easily tell these two methods do the same thing "in reverse"

Reply:
Yes that makes sense, I went with `encodeOptions` now to remain consistent with the overall naming: d35f44441.